libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/model/builder/KotlinModelBuilder.kt

```diff
@@ -91,6 +101,33 @@ class KotlinModelBuilder(private val kotlinPluginVersion: String) : ToolingModel
             } else null
         }
 
+        /**
+         * The [SourceSet] returned by this method is not intended to be complete, most of the information here is exposed by the
+         * Android Gradle plugin and as such is not populated here. The important information here that is required by
+         * Android Studio are the [CompilerArguments].
+         */
+        private fun AbstractKotlinCompile<*>.createAndroidSourceSet(androidTarget: KotlinAndroidTarget): SourceSet {
+            val variantName = sourceSetName
+            val compilation = androidTarget.compilations.getByName(variantName)
+            // Merge all sources and resource dirs from the different Source Sets that make up this variant.
+            val sources = compilation.kotlinSourceSets.flatMap {
```

Comment:
Use [`allKotlinSourceSets`](https://github.com/JetBrains/kotlin/blob/a44d70e79d06a5133b86470c749cc2d0bf870053/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/mpp/kotlinCompilations.kt#L166) instead. The difference between them is that `kotlinSourceSets` only contains the source sets added directly to the compilation, while `allKotlinSourceSets` also includes source sets that are transitively added to the compilation through the `dependsOn` relation.

Reply:
Changed to allKotlinSourceSets